runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineTranslator.java

```diff
@@ -734,10 +731,10 @@ private void addDisplayData(String stepName, HasDisplayData hasDisplayData) {
       } catch (Exception e) {
         String msg = String.format("Exception thrown while collecting display data for step: %s. "
             + "Display data will be not be available for this step.", stepName);
-        DisplayDataException displayDataException = new DisplayDataException(msg, e);
+        RuntimeException displayDataException = new RuntimeException(msg, e);
         LOG.warn(msg, displayDataException);
 
-        displayData = displayDataException.asDisplayData();
+        displayData = DisplayData.from(displayDataException);
```

Comment:
Seems like a good idea to me: that specializes the API which is good since this is a niche scenario.